<a href="https://colab.research.google.com/github/MaicolHernandez33/Actividad1/blob/main/Actividad1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [81]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

df=pd.read_csv('1.1.6. Archivo para actividad práctica.csv')
df.head(15)


,S.NO,Name,Nationality,Current Rank,Previous Year Rank,Sport,Year,earnings ($ million)
0,1,Mike Tyson,USA,1,NaN,boxing,1990,28.6
1,2,Buster Douglas,USA,2,NaN,boxing,1990,26.0
2,3,Sugar Ray Leonard,USA,3,NaN,boxing,1990,13.0
3,4,Ayrton Senna,Brazil,4,NaN,auto racing,1990,10.0
4,5,Alain Prost,France,5,NaN,auto racing,1990,9.0
5,6,Jack Nicklaus,USA,6,NaN,golf,1990,8.6
6,7,Greg Norman,Australia,7,NaN,golf,1990,8.5
7,8,Michael Jordan,USA,8,NaN,basketball,1990,8.1
8,9,Arnold Palmer,USA,8,NaN,golf,1990,8.1
9,10,Evander Holyfield,USA,8,NaN,boxing,1990,8.1


1. Cargue los datos en un dataframe
2. Identificar el tipo de datos de cada variable.
3. Calcular estadísticas básicas para cada variable. Sea de posición, dispersión y tendencia central. Para el caso de variables categóricas
calcule la moda.
4. Transforme la variable “earnings” en una variable categórica implementando intervalos de clase. Calcule la frecuencia de cada clase.
5. Conteste las siguientes preguntas:  
a) ¿Quién es la persona que más dinero ganó en un año?  
b) ¿Cuál es el deporte que más se repite dentro de la lista?  
c) ¿Cuál es el deporte donde más dinero se gana?  
d) Si quisiera apostar a un jugador de Basketball según su ranking. ¿A cuál debería apostar?  
e) ¿Para qué áreas de negocio puede ser útil el análisis de este dataset? ¿Qué decisiones puede ayudar a tomar?  
6. Entregue conclusiones sobre lo desarrollado. Comente los descubrimientos que más le sorprendieron, lo que más le costó y qué
otra información le gustaría extraer de los datos trabajados

## 2. Tipos de dato:  
  
- Name: Cualitativo nominal
- Nationality: Cualitativo nominal
- Current Rank: Cuantitativa discreta
- Previous Year Rank: Cuantitativa discreta
- Sport: Cualitativo nominal
- Year: Cuantitativa discreta
- earnings ($ million): Cuantitativa continua


## 3. Variables Categóricas

In [ ]:
df['Name'].mode()

,Name
0,Michael Jordan
1,Tiger Woods


En este caso la moda serian 2, Michel Jordan y Tiger Woods siendo los nombres que más se repiten en la bd.

In [ ]:
df['Nationality'].mode()

,Nationality
0,USA


Para la nacionalidad la moda seria USA siendo el que más se repite de la naciodanidad de los deportistas.

In [ ]:
df['Sport'].mode()

,Sport
0,Basketball


La moda de la categoría 'Sport' es Basketball

In [ ]:
df['Previous Year Rank'].mode()

,Previous Year Rank
0,>10


La moda de 'Previous Year Rank' es ">10", indicando que la mayor frecuencia corresponde a deportistas que en el año anterior estaban fuera del top 10 del ranking.
  
De esta categoría solo hice la moda ya que aunque tenia valores numéricos también tenía textos o datos como >30, siendo de momento dificil sacar estadisticas basicas de aca ya que habria que limpiar o transformar algunos datos como los NaN o >30.

## 3. Variables númericas

In [56]:
print("Esatadísticas de 'Current Rank'\n")
print(f"Media: {df['Current Rank'].mean():.2f}")
print(f"Mediana: {df['Current Rank'].median():.2f}")
print(f"Desviación estándar: {df['Current Rank'].std():.2f}")
print(f"Varianza: {df['Current Rank'].var():.2f}")
print(f"Mínimo: {df['Current Rank'].min():.2f}")
print(f"Máximo: {df['Current Rank'].max():.2f}")

Esatadísticas de 'Current Rank'

Media: 5.45
Mediana: 5.00
Desviación estándar: 2.85
Varianza: 8.13
Mínimo: 1.00
Máximo: 10.00


La media del Current Rank es 5.45, lo que indica que en promedio los deportistas se encuentran alrededor de la quinta posición del ranking. La mediana es 5, lo que confirma ya que no esta muy lejos de lo que nos dio el promedio.
  
La desviación estándar de 2.85 muestra que las posiciones no varían demasiado dentro del ranking, lo cual es lógico ya que el listado está limitado entre el puesto 1 y el 10. El valor mínimo corresponde al primer lugar, mientras que el máximo es el puesto 10.

In [57]:
print("Esatadísticas de 'Year'\n")
print(f"Media: {df['Year'].mean():.2f}")
print(f"Mediana: {df['Year'].median():.2f}")
print(f"Desviación estándar: {df['Year'].std():.2f}")
print(f"Varianza: {df['Year'].var():.2f}")
print(f"Mínimo: {df['Year'].min()}")
print(f"Máximo: {df['Year'].max()}")

Esatadísticas de 'Year'

Media: 2005.12
Mediana: 2005.00
Desviación estándar: 9.06
Varianza: 82.15
Mínimo: 1990
Máximo: 2020


La media del año es 2005.12, lo que indica el punto medio de los datos. La mediana también es 2005 confirmando lo anterior siendo practicamente igual.

La desviación estándar de 9.06 años indica que los datos abarcan un período bastante amplio.

In [58]:
print("Esatadísticas de 'earnings ($ million)'\n")
print(f"Media: {df['earnings ($ million)'].mean():.2f}")
print(f"Mediana: {df['earnings ($ million)'].median():.2f}")
print(f"Desviación estándar: {df['earnings ($ million)'].std():.2f}")
print(f"Varianza: {df['earnings ($ million)'].var():.2f}")
print(f"Mínimo: {df['earnings ($ million)'].min()}")
print(f"Máximo: {df['earnings ($ million)'].max()}")

Esatadísticas de 'earnings ($ million)'

Media: 45.52
Mediana: 39.00
Desviación estándar: 33.53
Varianza: 1123.95
Mínimo: 8.1
Máximo: 300.0


La media de ganancias es de 45.52 millones de dólares, lo que representa el ingreso promedio anual de los deportistas, en el caso de la mediana es 39 millones, lo que nos indica que hay valores altos que estan inflando la media.
  
La desviación estándar de 33.53 muestra que existe una gran variabilidad en las ganancias entre los deportistas. El rango es amplio, con valores que van desde 8.1 millones hasta 300 millones de dólares, esto nos indica que hay mucha diferencia entre lo que ganan los diferentes deportistas.

## 4. Tranformacion de Variable 'earnings ($ million)'

In [59]:
df['earnings_category'] = pd.cut(df['earnings ($ million)'], bins=5)

In [62]:
df['earnings_category'].value_counts()

,count
earnings_category,
"(7.808, 66.48]",243
"(66.48, 124.86]",54
"(124.86, 183.24]",2
"(241.62, 300.0]",2
"(183.24, 241.62]",0



La variable 'earnings ($ million)' la transformamos en una variable categórica en intervalos de clases para poder analizar las ganancias juntando los valores en rangos de dinero. Despues se calcula la frecuencia de cada intervalo para ver en que rango se encuentra la mayor cantidad de deportistas.

## 5. Conteste las siguientes preguntas

a) ¿Quién es la persona que más dinero ganó en un año?   
R: La persona que más dinero ganó en un año fue Floyd Mayweather Jr., con aproximadamente 300 millones de dólares en ganancias.
    
b) ¿Cuál es el deporte que más se repite dentro de la lista?    
R: El deporte que más se repite dentro de la lista es basketball, ya que corresponde al valor más frecuente (moda) de la variable Sport.
    
c) ¿Cuál es el deporte donde más dinero se gana?  
R: El deporte donde más dinero se gana es boxing, ya que presenta el mayor promedio de ganancias entre los deportes del dataset.
  
d) Si quisiera apostar a un jugador de Basketball según su ranking. ¿A cuál debería apostar?     
R: Si se quisiera apostar por un jugador de basketball basándose en su ranking, la mejor opción sería Michael Jordan, ya que aparece con las mejores posiciones dentro del ranking de ganancias del dataset.
  
e) ¿Para qué áreas de negocio puede ser útil el análisis de este dataset? ¿Qué decisiones puede ayudar a toma    
R: El análisis de este dataset puede ser útil para diversas áreas de negocio relacionadas con la industria del deporte. Por ejemplo: Marketing deportivo, Patrocinios, medios, entretenimiento y gestión deportiva
  
Este tipo de análisis puede ayudar a tomar decisiones como: Qué deportistas contratar para campañas publicitarias, en qué deportes invertir más dinero, qué deportes generan mayores ingresos a lo largo del tiempo, etc.

In [71]:
df.loc[df['earnings ($ million)'].idxmax()]

,241
S.NO,242
Name,Floyd Mayweather
Nationality,USA
Current Rank,1
Previous Year Rank,1
Sport,Boxing
Year,2015
earnings ($ million),300.0


In [72]:
df['Sport'].mode()

,Sport
0,Basketball


In [76]:
df.groupby('Sport')['earnings ($ million)'].mean()

,earnings ($ million)
Sport,
American Football,53.011765
American Football / Baseball,22.500000
Auto Racing,25.480000
Auto Racing (Nascar),22.250000
Auto racing,18.500000
Baseball,18.633333
Basketball,45.879630
Boxing,60.110345
F1 Motorsports,49.600000


In [77]:
df[df['Sport'] == 'basketball'].sort_values('Current Rank').head(1)

,S.NO,Name,Nationality,Current Rank,Previous Year Rank,Sport,Year,earnings ($ million)
50,51,Michael Jordan,USA,1,1,basketball,1995,43.9


## 6. Conclusiones sobre lo desarrollado.


A partir de los diferentes analisis y estadísticas que sacamos se pueden identificar varias cosas interesantes como por ejemplo que hubieron dos deportistas que se repitieron bastantes siendo la moda estos, esto indica que se mantivieron ganando bastante dinero durante varios años, a demas de sacar la moda de que sport era el mas recurrente, este caso fue el basketball, que nos indica que en este deporte se movio mucho dinero en estos años.
  
Otra cosa que me llamo la atencion fue la diferencia de ganancia de los deportistas siendo lo mínimo 8 millones y lo máximo 300 millones dando una diferencia muy grande. Esto nos dice que hay una gran variabilidad ecónomica entre los deportistas. También vimos que esos datos tan alto hacen que el promedio suba bastante ya que al sacar la mediana se notaba que el promedio había subido bastante por estos datos.

La parte que más me dificulto fue sobre la pregunta: Si quisiera apostar a un jugador de Basketball según su ranking. ¿A cuál debería apostar?, esto no lo pude hacer y tuve que buscar en internet, porque con la moda sabía cual era el deportista que mas se repetia pero como pedia según su rankin no pude hacerlo. Otra cosa que me dificulto fue la parte 4 donde se tuvo que transformar la variable.

Pero en general estuvo bastante bien, se podía y aun se puede analizar muchas mas cosas de los datos que tenemos en el dataset.